backend/compact-connect/stacks/persistent_stack/compact_configuration_upload.py

```diff
@@ -0,0 +1,173 @@
+        """
+        Check if the compact configuration is active in the given environment.
+        """
+        return environment_name in active_environments or self.node.try_get_context('sandbox') == 'true'
+
+    def _generate_compact_configuration_json_string(self, environment_name: str) -> str:
+        """
+        Currently, all configuration for compacts and jurisdictions is hardcoded in the compact-config directory.
+        This reads the YAML configuration files and generates a JSON string of all the configuration objects that
+        should be uploaded into the provider table.
+        """
+
+        # This object consists of two attributes: compacts and jurisdictions.
+        # 'compacts' is a list of top level compact configuration files converted into JSON.
+        # 'jurisdictions' is a dictionary of jurisdiction configuration files for a specific
+        # compact, converted into JSON.
+        uploader_input = {
+            "compacts": [],
+            "jurisdictions": {}
+        }
+
+        # Read all compact configuration YAML files from top level compact-config directory
+        for compact_config_file in os.listdir('compact-config'):
+            if compact_config_file.endswith('.yml'):
+                with open(os.path.join('compact-config', compact_config_file), 'r') as f:
+                    # convert YAML to JSON
+                    formatted_compact = yaml.safe_load(f)
+                    # only include the compact configuration if it is active in the environment
+                    if self._configuration_is_active_for_environment(environment_name,
+                                                                     formatted_compact['activeEnvironments']):
+                        uploader_input['compacts'].append(formatted_compact)
+
+        # Read all jurisdiction configuration YAML files from each active compact directory
+        for compact in uploader_input['compacts']:
+            compact_name = compact['compactName']
+            uploader_input['jurisdictions'][compact_name] = []
+            for jurisdiction_config_file in os.listdir(os.path.join('compact-config', compact['compactName'])):
+                if jurisdiction_config_file.endswith('.yml'):
+                    with open(os.path.join('compact-config', compact_name, jurisdiction_config_file), 'r') as f:
+                        # convert YAML to JSON
+                        formatted_jurisdiction = yaml.safe_load(f)
+                        # only include the jurisdiction configuration if it is active in the environment
+                        if self._configuration_is_active_for_environment(environment_name,
+                                                                         formatted_jurisdiction['activeEnvironments']):
+                            uploader_input['jurisdictions'][compact_name].append(formatted_jurisdiction)
```

Comment:
Is there a way to 'test' this locally? e.g., I can imagine misspelling a compact directory and not noticing, and it might be nice if a test would print the active compact and jurisdiction configurations under a particular environment.

Reply:
The way you would test this locally is by running the CDK synth for the persistent stack and then checking the CFN template that is generated. The template will include the JSON formatted string that is generated and being sent to the custom resource.

Reply:
Got it. Helpful! Is that worth documenting in the markdown file above? (It might not be given the intended audience, genuine question. Your call!)

Reply:
That's a good question... it might be helpful to briefly mention that making any changes to the files may break the snapshot tests. So yes, I'll add a brief testing section in there.